SshAgentLib/PpkFormatter.cs

```diff
@@ -379,6 +483,49 @@ public override object Deserialize(Stream aStream)
 
     #region -- Private Methods --
 
+    private static void argonKeys(FileData fileData, out byte[] cipherKey, out byte[] iv, out byte[] macKey)
+    {
+      int cipherLength = 32;
+      int ivLength = 16;
+      int macLength = 32;
+
+      if (fileData.privateKeyAlgorithm == PrivateKeyAlgorithm.None) {
+        cipherKey = null;
+        iv = null;
+        macKey = new byte[0];
+        return;
+      }
+      using (var passphrase = fileData.passphrase.ToAnsiArray()) {
+        Argon2 hasher;
+        switch (fileData.kdfAlgorithm) {
+          case KeyDerivation.Argon2i:
+            hasher = new Argon2i(passphrase.Data);
+            break;
+          case KeyDerivation.Argon2d:
+            hasher = new Argon2d(passphrase.Data);
+            break;
+          case KeyDerivation.Argon2id:
+            hasher = new Argon2id(passphrase.Data);
+            break;
+          default:
+            throw new ArgumentOutOfRangeException();
+        }
+        hasher.MemorySize = (int) fileData.kdfParameters[argonMemoryKey];
+        hasher.Iterations = (int) fileData.kdfParameters[argonPassesKey];
+        hasher.DegreeOfParallelism = (int) fileData.kdfParameters[argonParallelismKey];
+        hasher.Salt = (byte[]) fileData.kdfParameters[argonSaltKey];
+
+        // These values are copied by Aes and HMACSHA256 which
+        // means they aren't explicitly zeroed unless we do it.
+        // and then cipher.Clear() and mac.Clear() need to be
+        // called once they're no longer in use.
+        byte[] kdf = hasher.GetBytes(cipherLength + ivLength + macLength);
+        cipherKey = kdf.Skip(0).Take(cipherLength).ToArray();
```

Comment:
Can `Skip(0)` be omitted?

Reply:
Yes it can, but I prefer to keep it that way because I like the offset and length to be explicit.

Reply:
ok